File: examples/lua/client/protocols/ReqRoleCreate.lua
module("ReqRoleCreate", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqRoleCreate})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.uid = 0
	self.uuid = 0
	self.sid = 0
	self.cid = 0
	self.os = 0
	self.version = 0
	self.uname = 0
	self.source = 0
	self.source_sub = 0
	self.login_time = 0
end


function encode(self)
	 local pack = packet.new()
	pack:writeU32(self.uid)
	pack:writeU32(self.uuid)
	pack:writeU16(self.sid)
	pack:writeU16(self.cid)
	pack:writeString(self.os)
	pack:writeString(self.version)
	pack:writeString(self.uname)
	pack:writeString(self.source)
	pack:writeString(self.source_sub)
	pack:writeU32(self.login_time)
	return pack:encode(Msg.P_REQ_ROLE_CREATE)
end

File: examples/golang/server/proto/req_role_create.go
package proto

import (
	"packet"
)

type ReqRoleCreate struct {
	Uid                      uint32
	Uuid                     uint32
	Sid                      uint16
	Cid                      uint16
	Os                       string
	Version                  string
	Uname                    string
	Source                   string
	SourceSub                string
	LoginTime                uint32
}

func ReqRoleCreateDecode(pack *packet.Packet) *ReqRoleCreate {
	reqRoleCreate := &ReqRoleCreate{}

	reqRoleCreate.Uid = pack.ReadUint32()
	reqRoleCreate.Uuid = pack.ReadUint32()
	reqRoleCreate.Sid = pack.ReadUint16()
	reqRoleCreate.Cid = pack.ReadUint16()
	reqRoleCreate.Os = pack.ReadString()
	reqRoleCreate.Version = pack.ReadString()
	reqRoleCreate.Uname = pack.ReadString()
	reqRoleCreate.Source = pack.ReadString()
	reqRoleCreate.SourceSub = pack.ReadString()
	reqRoleCreate.LoginTime = pack.ReadUint32()
	return reqRoleCreate
}
